versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/RegLogSpliterator.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.nontx;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Spliterators.AbstractSpliterator;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.RefLogNotFoundException;
+import org.projectnessie.versioned.persist.adapter.RefLog;
+
+/**
+ * {@link Spliterator} of {@link RefLog} entries using a variable number of ref-log stripes,
+ * according to {@link NonTransactionalDatabaseAdapterConfig#getRefLogStripes()} to read {@link
+ * RefLog}s from in parallel.
+ *
+ * <p>The next {@link RefLog} returned by this implementation is the next available from all stripes
+ * with the lowest {@link RefLog#getOperationTime()}.
+ */
+final class RegLogSpliterator extends AbstractSpliterator<RefLog> {
+
+  private final List<RefLogSplit> splits;
+
+  private final Hash initialHash;
+  private boolean initialHashSeen;
+  private RefLog initialRefLog;
+
+  RegLogSpliterator(Hash initialHash, Stream<Spliterator<RefLog>> refLogStripeFetcher)
+      throws RefLogNotFoundException {
+    super(Long.MAX_VALUE, 0);
+
+    this.initialHash = initialHash;
+    this.initialHashSeen = initialHash == null;
+
+    splits = refLogStripeFetcher.map(RefLogSplit::new).collect(Collectors.toList());
+
+    // The API for DatabaseAdapter.refLog(Hash) requires to throw a RefLogNotFoundException,
+    // if no RefLog for the initial hash exists. This loop tries to find the initial RefLog entry
+    // and throw
+    AtomicReference<RefLog> initial = new AtomicReference<>();
+    while (!initialHashSeen) {
+      if (!tryAdvance(initial::set)) {
+        break;
```

Comment:
Could we have `tryAdvance` in the `while` condition and break of receiving the initial entry? I think the code would be more readable that way. WDYT (see also the comment on line 100)?